### DISCUSSION TITLE
What is necessary to create a new table based on the Table class?

### DISCUSSION THREAD

**Discussion by mdissel** [JasperFx/marten#1676] [Q&A]
In my situation i need to create new table to store simple / plain objects without any json (each property has it's own column). 
So far i have:
- a Table implementation with my columns. Adding an index in the constructor to seems a bit strange due the DocumentMapping requirement.
`Indexes.Add(new IndexDefinition(DocumentMapping.For<ListContact>(), "search") { Method = IndexMethod.gin, Expression = "\"search\" gin_trgm_ops", IsConcurrent = true });`
- an IFeatureSchema that's returning the Table instance
- where can I register the upsert sql functions? I expected that to be in the IFeatureSchema..
- what is necessary to to support session.Store(new ListContact()) (without a 'manual' IStorageOperation) and session.Query<ListContact>

Thanks

**Comment by jeremydmiller** [comment]
Sorry for being slow here. Look at the `IFeatureSchema` abstraction in the Marten codebase. Somewhere, there's a way to add new ones of those to the `StoreOptions` that will get enlisted in Marten's schema object management. Other folks have done that, and I've used that at one point myself.
